colonist/processor/src/test/java/com/joom/colonist/processor/integration/IntegrationTestRule.kt

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2022 SIA Joom
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.joom.colonist.processor.integration
+
+import com.joom.colonist.processor.ColonistParameters
+import com.joom.colonist.processor.ColonistProcessor
+import com.joom.colonist.processor.ErrorReporter
+import com.joom.colonist.processor.ProcessingException
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.PrintStream
+import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.exists
+import org.jetbrains.kotlin.cli.common.ExitCode
+import org.jetbrains.kotlin.cli.common.arguments.K2JVMCompilerArguments
+import org.jetbrains.kotlin.cli.common.messages.MessageRenderer
+import org.jetbrains.kotlin.cli.common.messages.PrintingMessageCollector
+import org.jetbrains.kotlin.cli.jvm.K2JVMCompiler
+import org.jetbrains.kotlin.config.Services
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+class IntegrationTestRule(
+  rootPackage: String,
+  private val testCaseProjectsDir: Path = COMPILED_TEST_CASE_PROJECTS_PATH
+) : TestRule {
+  private val rootDirectory = rootPackage.replace(".", File.separator)
+
+  private val compiledFilesDirectory = testCaseProjectsDir.resolve("_generated")
+  private val processedDirectory = testCaseProjectsDir.resolve("_processed")
+  private val classpath = JvmRuntimeUtil.computeRuntimeClasspath()
+
+  fun assertValidProject(sourceCodeDir: String, discoverSettlers: Boolean = true) {
+    val reporter = ErrorReporter()
+    processProject(sourceCodeDir, reporter, discoverSettlers = discoverSettlers)
+
+    reporter.shouldNotHaveErrors()
+  }
+
+  fun assertInvalidProject(sourceCodeDir: String, message: String, discoverSettlers: Boolean = true) {
+    val reporter = ErrorReporter()
+    processProject(sourceCodeDir, reporter, discoverSettlers = discoverSettlers, ignoreErrors = true)
+
+    reporter.shouldContainErrorMessage(message)
+  }
+
+  fun compileProject(sourceCodeDir: String, classpath: List<Path> = emptyList()): Path {
```

Comment:
Minor: some functions below can be private?

Reply:
I am planning to add tests that'll use these